File: Question 1/Bug_rearrange.c
#include <stdio.h>

int* rearrange(int *arr, int n) {

    /*int temp[n];
    for (int i = 0; i < n; i++) {
        temp[i] = arr[arr[i]];
    }

    for (int i = 0; i < n; i++) {
        arr[i] = temp[i];
    }*/

	for(int i = 0; i < n; i++)
		arr[i] = arr[i] + arr[arr[i]] * n;
		
	for(int i = 0; i < n; i++)
		arr[i] = arr[i] * 1.0 / n;
    
	// *result_count = n;
    return arr;
}

int main(){
	int n;
	scanf("%d", &n);

	int a[n];
	for(int i = 0;i<n;i++){
		scanf("\n%d ", &a[i]);
	}

	rearrange(a,n);
		
    printf("\n");
	for(int i = 0;i<n;i++)
		printf("%d ", a[i]);
    printf("\n");
}

File: Question 1/Bug_rearrange.py
#!/bin/python3

def rearrange(arr, n):
    
    for i in range(n):
        arr[i] = arr[i] + arr[arr[i]] * n

    for i in range(n):
        arr[i] /= n

    return arr

if __name__ == "__main__":
    n = int(input())
    arr = list(map(int, input().split()))

    result = rearrange(arr, n)

    print(result)
